### PR TITLE
Add the security hud upgrade for cyborgs

### PR DESCRIPTION
**Pull request by Luxizzle**
[FEATURE]

## About the PR 

Adds a new Security HUD upgrade for cyborgs, with it they can see the same icons security can see with security glasses. Replaces the repair pack in brobocop to the new upgrade and adds it to the robotics fabricator.

## Why's this needed? 

Playing a bit of brobocop lately and while assisting security, one thing that i noticed that was very much missing is the ability to see arrest statuses. So this adds them!

I know there was a security module at some point and I'm not sure if that ever included this, but i don't think it is that overpowered to give to cyborgs.
I only play on the roleplay servers though, i have no context on how the normal servers function in terms of balance.

## Changelog

```
(u)Luxizzle:
(*)Added the Security HUD upgrade for cyborgs.
(+)Replaced Repair Pack in brobocop module to the new Security HUD upgrade.
```

### CODE REVIEW

File: code/modules/robotics/robot/upgrade/sechud_scanner.dm [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	desc = "An advanced sensor array that allows a cyborg to quickly determine the physical condition of organic life."
+	icon_state = "up-sechud"
+	drainrate = 5
+
+	var/client/assigned = null
+
+	proc/updateIcons()
+		// I wouldve liked to avoid this but i dont want to put this inside the mobs life proc as that would be more code.
+		// TODO: signals/components!
+		while (src.assigned)
```
Comment (ZeWaka) [inline code/modules/robotics/robot/upgrade/sechud_scanner.dm:12]: I would subscribe to item processing and use that instead, this is not good and will never stop even if it's outside a bot

File: code/modules/robotics/robot/upgrade/sechud_scanner.dm [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+				if (I.loc.invisibility && I.loc != src.loc)
+					continue
+				src.assigned.images.Add(I)
+
+	upgrade_activate(var/mob/living/silicon/robot/user as mob)
+		if (..())
+			return
+		src.assigned = user.client
+		SPAWN_DBG(-1)
+			updateIcons()
```
Comment (ZeWaka) [inline code/modules/robotics/robot/upgrade/sechud_scanner.dm:36]: yeah not good

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Comment by Luxizzle** [comment]
This is code copied from the prodoc upgrade, i'll see if i can get it fixed over there too

File: code/modules/robotics/robot/upgrade/sechud_scanner.dm (hunk 1 above)
Comment (Luxizzle) [reply code/modules/robotics/robot/upgrade/sechud_scanner.dm:12]: Got a small example or somewhere i could find where something like this is used? Both sechud and prodocs seem to just work on the process loop.

**Comment by Luxizzle** [comment]
Alright, using processing now. Never used this so if i did anything wrong please yell at me.

File: code/modules/robotics/robot/upgrade/sechud_scanner.dm [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		src.assigned = user.client
+		processing_items |= src
+
+	upgrade_deactivate(var/mob/living/silicon/robot/user as mob)
+		if (..())
+			return
+		if (src.assigned)
+			src.assigned.images.Remove(arrestIconsAll)
+			src.assigned = null
+			processing_items.Remove(src)
```
Comment (ZeWaka) [inline code/modules/robotics/robot/upgrade/sechud_scanner.dm:42]: couldn't the client null out, and then it'd never get removed from processing items?

File: code/modules/robotics/robot/upgrade/prodoc_scanner.dm [hunk 4]
```diff
@@ -32,13 +31,13 @@
...
-		updateIcons()
+	processing_items |= src
 
 /obj/item/roboupgrade/healthgoggles/upgrade_deactivate(var/mob/living/silicon/robot/user as mob)
 	if (..())
 		return
 	if (src.assigned)
 		src.assigned.images.Remove(health_mon_icons)
 		src.assigned = null
+		processing_items.Remove(src)
```
Comment (ZeWaka) [inline code/modules/robotics/robot/upgrade/prodoc_scanner.dm:42]: ditto

File: code/modules/robotics/robot/upgrade/sechud_scanner.dm (hunk 3 above)
Comment (Luxizzle) [reply code/modules/robotics/robot/upgrade/sechud_scanner.dm:42]: Doesn't the else in the process proc after checking src.assigned take care of that though?

**Review by ZeWaka — APPROVED** [review]

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**